test/integration/framework/apiserver.go

```diff
@@ -121,31 +120,47 @@ func readCustomResourcesAtPath(t *testing.T, path string) []*v1beta1.CustomResou
 		if filepath.Ext(path) != ".yaml" {
 			return nil
 		}
-		crd, err := readCRDAtPath(codec, converter, path)
+		crd, err := readCRDsAtPath(codec, converter, path)
 		if err != nil {
 			return err
 		}
-		crds = append(crds, crd)
+		crds = append(crds, crd...)
 		return nil
 	}); err != nil {
 		t.Fatal(err)
 	}
 	return crds
 }
 
-func readCRDAtPath(codec runtime.Codec, converter runtime.ObjectConvertor, path string) (*v1beta1.CustomResourceDefinition, error) {
+func readCRDsAtPath(codec runtime.Codec, converter runtime.ObjectConvertor, path string) ([]*v1beta1.CustomResourceDefinition, error) {
 	data, err := ioutil.ReadFile(path)
 	if err != nil {
 		return nil, err
 	}
 
-	internalCRD := &apiextensions.CustomResourceDefinition{}
-	if _, _, err := codec.Decode(data, nil, internalCRD); err != nil {
-		return nil, err
+	dataStr := string(data)
+	datas := strings.Split(dataStr, "---")
```

Comment:
is it maybe an idea to split on `\n---\n` to not cause any `---` to make it split the file